### DISCUSSION TITLE
Should we update to the latest version of the LSP spec?

### DISCUSSION THREAD

**Discussion by PurpleMyst** [PurpleMyst/sansio-lsp-client#19] [General]
As per the title, this library explicitly supports LSP3.0 and we're on 3.16, almost 3.17

What should be our uh, upgrade path? We can't just pull the rug under our users and stop supporting 3.0 if there's breaking changes, but newest versions of the spec have *very* interesting features like code highlighting et cetera.

We should check which features could actually be useful, especially within porcupine, and decide from there.

What do you think, @Akuli?

**Comment by Akuli** [comment]
Is there a concise "what's new" for changes from what we support to the latest langserver protocol?

**Reply by PurpleMyst** [reply]
[This should be it](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#version_3_0_0).

**Comment by Akuli** [comment]
I think the supported features should be determined by what Porcupine users want (or users of any other editors using sansio-lsp-client). To me, that is mostly autocompletions and diagnostics, although "jump to definition" and "find usages" would be nice. If someone else requests a feature on e.g. Porcupine issue tracker, we should consider supporting that too.

In #21, I discovered that clangd doesn't work. We should fix that next.

**Reply by PurpleMyst** [reply]
That's valid.
